File: akka-mini-cluster-protocol/src/main/java/albgorski/akka/protocol/GreeterProtocol.java
package albgorski.akka.protocol;

import java.io.Serializable;

public abstract class GreeterProtocol {

    public static final class RequestHello implements Serializable {
        public final String from;

        public RequestHello(String from) {
            this.from = from;
        }

        @Override
        public String toString() {
            return "RequestHello{" +
                    "from='" + from + '\'' +
                    '}';
        }
    }
}

File: akka-mini-cluster-protocol/src/main/scala/albgorski/akka/protocol/GreeterProtocol.scala
package albgorski.akka.protocol


case class RequestHelloScalaVersion(from: java.lang.String, year: Integer)

object RequestHello {
  import GreeterProtocol.RequestHello

  def apply(from: String) =
    new RequestHello(from)

  def unapply(user: RequestHello) =
    Some(user.from)

  implicit class RequestHelloScalaApi(user: RequestHello) {
  }

}